src/index.js

```diff
@@ -4,28 +4,51 @@ function updateSubtotal(product) {
   console.log('Calculating subtotal, yey!');
 
   //... your code goes here
+  const price = product.querySelector('.price span');
+  const quantity = product.querySelector('.quantity input');
+  
+  let total = price.innerHTML * quantity.value;
+
+  let subtotal = product.querySelector('.subtotal span');
+
+  subtotal.innerHTML = total;
+  return total;
 }
 
 function calculateAll() {
   // code in the following two lines is added just for testing purposes.
   // it runs when only iteration 1 is completed. at later point, it can be removed.
-  const singleProduct = document.querySelector('.product');
-  updateSubtotal(singleProduct);
+  /*const singleProduct = document.querySelector('.product');
+  updateSubtotal(singleProduct);*/
   // end of test
 
   // ITERATION 2
   //... your code goes here
+  let allProducts = document.getElementsByClassName("product");
+  allProducts = [...allProducts];
+
+  let total = 0;
+  for (let i = 0; i < allProducts.length ; i++){
+    total += updateSubtotal(allProducts[i]);
+  }
 
   // ITERATION 3
   //... your code goes here
+  let allTotal = document.querySelector('#total-value span');
+  allTotal.innerHTML = total ;
+
 }
 
 // ITERATION 4
 
 function removeProduct(event) {
   const target = event.currentTarget;
-  console.log('The target in remove is:', target);
+  console.log('The target in remove is:', target.parentNode.parentNode);
+  target.parentNode.parentNode.remove();
+
```

Comment:
You correctly created the function to remove a product, however, you are not telling your code to "listen" for the click to trigger the action (call the function to remove the determined product).
Just like you have inside `window.addEventListener()` something that triggers the `calculateAll` function when there's a click in an element with the id `calculate`, you needed to add something similar to this, for when the user clicks to remove the product.
Here's a possible solution for you to add inside that event listener:
 ```
const buttonRemove = document.getElementsByClassName("btn-remove");
  for (let i = 0; i < buttonRemove.length; i++) {
    buttonRemove[i].addEventListener("click", removeProduct);
  }
```